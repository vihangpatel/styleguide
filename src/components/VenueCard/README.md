Venue Card:

```js
    <VenueCard
        imageUrl='//in.bmscdn.com/venueimages/venuelisting/GEKE.jpg'
        title='Actor Prepares: Andheri (West)'
        address='Mukti Building, 5th Floor, 141 Model Town, Behind Kokilaben Ambani Hospital, Four Bungalows, Next to Levo Restaurant and Lounge, Mumbai, Maharashtra 400053, India'
        distance='1.5 KM' />
```

Loading Venue Card:
```js
    <VenueCard isLoading />
```
