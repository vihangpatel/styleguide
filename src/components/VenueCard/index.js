import React from 'react'
import PropTypes from 'prop-types'
import { createComponent } from 'react-fela'
import { margin } from 'polished'
import Image from './components/Image'
import Favorite from './components/Favorite'
import Info from './components/Info'
import VenuePlaceHolder from './components/VenuePlaceHolder'

const __card = props => ({
	...margin('8', 'auto'),
	backgroundColor: '#fff',
	display: 'flex',
	alignItems: 'flex-start',
	border: props.theme.border,
	borderRadius: props.theme.borderRadius,
	mobileXs: {
		width: `${props.lockWidth ? '270' : 'auto'}`,
	},
	mobileX: {
		width: `${props.lockWidth ? '320' : 'auto'}`,
	},
	height: props.height || 'auto',
	...props.theme.regularBaseFont,
	position: 'relative',
	overflow: 'hidden',
	extend: {
		condition: props.horizontalScroll,
		style: {
			...margin('8'),
			mobileXs: {
				minWidth: `${props.lockWidth ? '270' : 'auto'}`,
				height: 'auto',
			},
			mobileX: {
				minWidth: `${props.lockWidth ? '320' : 'auto'}`,
				height: 'auto',
			},
		},
	},
})

const Card = createComponent(__card, 'div', ['onClick'])

/** Venue card to be rendered on the venues page. */
const Venue = props => {
	const {
		isLoading,
		imageUrl,
		distance,
		title,
		address,
		venueCode,
		favourite,
		horizontalScroll,
		SynopsisVenue,
	} = props
	const venueSelect =
		props.handleVenueSelect &&
		props.handleVenueSelect.bind(null, title, venueCode)
	const propsAvailable = imageUrl || title || address || favourite || false

	const favouriteSelect = event => {
		if (props.handleFavouriteClick)
			props.handleFavouriteClick(venueCode, title)
		event.stopPropagation()
	}

	return isLoading || !propsAvailable ? (
		<VenuePlaceHolder lockWidth={props.lockWidth} />
	) : (
		<Card
			onClick={venueSelect}
			lockWidth={props.lockWidth}
			horizontalScroll={horizontalScroll}
			height={props.height}
		>
			<Image imageUrl={imageUrl} title={title} />
			<Info
				title={title}
				address={address}
				SynopsisVenue={SynopsisVenue}
			/>
			<Favorite
				distance={distance}
				favouriteSelect={favouriteSelect}
				favourite={favourite}
			/>
		</Card>
	)
}

VenueCard.propTypes = {
	/** URL for the venue image */
	imageUrl: PropTypes.string,
	/** Title of the venue */
	title: PropTypes.string,
	/** Address of the venue */
	address: PropTypes.string,
	/** Distance between venue and user's location */
	distance: PropTypes.string,
	/** Toggle this to toggle between placeholder and actual venue card */
	isLoading: PropTypes.bool,
	/** Callback to the page. It is a function */
	handleVenueSelect: PropTypes.func,
	/** Callback to the page for favourite icons. It is a function */
	handleFavouriteClick: PropTypes.func,
	/** Toggle favourite venue icon */
	favourite: PropTypes.bool,
}

VenueCard.defaultProps = {
	imageUrl: '',
	title: '',
	address: '',
	distance: '',
	isLoading: false,
	lockWidth: true,
	height: '',
}

export default function VenueCard({ children, ...props }) {
	return <Venue {...props}>{children}</Venue>
}
