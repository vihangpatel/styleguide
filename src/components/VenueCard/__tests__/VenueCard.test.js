import React from 'react'
import VenueCard from 'components/VenueCard'
import StyleProvider from 'utils/Provider'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'

const venueCardProps = {
	key: 'PTHV',
	venueCode: 'PTHV',
	address:
		'20 Janki Kutir, Juhu Church Road, Mumbai, Maharashtra 400049, India',
	title: 'Prithvi Theatre',
	imageUrl: 'in.bmscdn.com/venueimages/venuecover/PTHV.jpg',
	distance: '100',
}

describe('VenueCard Testing', () => {
	it('shallow renders correctly', () => {
		const venueCard = shallow(<VenueCard {...venueCardProps} />)
		// console.log(venueCard)
		// TODO: Write more complex tests
		expect(false).toBe(true)
	})

	it('Creates venue card snapshot', () => {
		const venueCardSnapshot = renderer
			.create(
				<StyleProvider>
					<VenueCard {...venueCardProps} />
				</StyleProvider>
			)
			.toJSON()

		expect(venueCardSnapshot).toMatchSnapshot()
	})

	it('Should be clickable', () => {
		const clickSpy = jest.fn()
		const renderedComponent = mount(
			<StyleProvider>
				<VenueCard {...venueCardProps} handleVenueSelect={clickSpy} />
			</StyleProvider>
		)
		renderedComponent.simulate('click')
		expect(clickSpy).toHaveBeenCalled()
	})
})
