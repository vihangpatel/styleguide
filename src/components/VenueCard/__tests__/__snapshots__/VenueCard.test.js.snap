// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VenueCard Testing Creates venue card snapshot 1`] = `
<div
  className="a b c d e f g h i j k l m"
  onClick={undefined}
>
  <div
    className="n o p q r s t"
  >
    <div>
      <img
        alt="Prithvi Theatre"
        className="u as w x y at au av aw ax ay az"
        onClick={undefined}
        src="in.bmscdn.com/venueimages/venuecover/PTHV.jpg"
      />
    </div>
  </div>
  <div
    className="j k n p ac ae af ag"
  >
    <p
      className="j ah ai aj ak"
    >
      Prithvi Theatre
    </p>
    <p
      className="j k al am an ao ap ak aq ae ar ag"
    >
      20 Janki Kutir, Juhu Church Road, Mumbai, Maharashtra 400049, India
    </p>
  </div>
</div>
`;
